packages/calypso-e2e/src/lib/components/rewritten-domain-search-component.ts

```diff
@@ -144,4 +144,24 @@ export class RewrittenDomainSearchComponent {
 		// Now click the enabled button using dispatchEvent to handle issues with the environment badge staying on top of the button.
 		await Promise.all( [ continueButton.dispatchEvent( 'click' ), this.page.waitForNavigation() ] );
 	}
+
+	/**
+	 * Skips the domain search screen.
+	 */
+	async skipPurchase(): Promise< string > {
+		const button = this.page.getByRole( 'button', { name: 'Skip purchase' } );
+
+		await button.waitFor();
+
+		let domain = await button.getAttribute( 'aria-label' );
+		domain = domain?.replace( 'Skip purchase and continue with ', '' ) ?? null;
```

Comment:
Is there a way to get the domain other than depending on `aria-label` and the fact that the label will always have this hardcoded English string?

Changing the label will force a few tests to fail for reasons that might not be clear to others.

Reply:
If the test fail that's a good thing, because then we're explicit about changing the behavior. E2E tests are supposed to behave like user interactions and that's one. I prefer that approach over data attributes to be found within tests.